packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -53,43 +47,71 @@ export const styles = {
   focusVisible: {},
 };
 
+const useEnhancedEffect = typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
```

Comment:
You might have already answered this point, it's hard for me to follow the past discussions in the GitHub UI. I'm wondering if we can end up in a weird case where the event is triggered but the handled hasn't been updated yet.
Why not follow the pattern of the linked comment? 

```suggestion
const useEnhancedEffect = React.useEffect;
```

Reply:
I we can do that, we can probably remove `unstable_useIsomorphicLayoutEffect` too.

Reply:
GitHub seemed to have lost part of my answer when it duplicated the conversation.

We can't use `useEffect` because their priority might be lower than user input events. Which means you could call a stale value instead. This has to be done synchronous i.e. with `useLayoutEffect`

Reply:
>Why not follow the pattern of the linked comment? 

I don't understand. The linked comment is not using `useEffect` but `useLayoutEffect`. The rest is just the usual "trick react into thinking this is OK for ssr".

Reply:
OK, my bad